Sources/App/configure.swift

```diff
@@ -7,13 +7,28 @@ public func configure(_ app: Application) throws {
     // uncomment to serve files from /Public folder
     // app.middleware.use(FileMiddleware(publicDirectory: app.directory.publicDirectory)){{#fluent}}
 
+    {{#fluent.db.is_postgres}}
+    let port: Int
+    if let portStr = Environment.get("DATABASE_PORT") {
+        guard let portInt = Int(portStr) else {
+            fatalError("DATABASE_PORT, if set, must be an integer value, not '\(portStr)'")
+        }
+
+        port = portInt
+    } else {
+        port = 5432
+    }
+    {{/fluent.db.is_postgres}}
+    
     {{#fluent.db.is_postgres}}app.databases.use(.postgres(
         hostname: Environment.get("DATABASE_HOST") ?? "localhost",
+        port: port,
         username: Environment.get("DATABASE_USERNAME") ?? "vapor_username",
         password: Environment.get("DATABASE_PASSWORD") ?? "vapor_password",
         database: Environment.get("DATABASE_NAME") ?? "vapor_database"
     ), as: .psql){{/fluent.db.is_postgres}}{{#fluent.db.is_mysql}}app.databases.use(.mysql(
         hostname: Environment.get("DATABASE_HOST") ?? "localhost",
+        port: port,
```

Comment:
I suggest doing it this way instead:
```suggestion
    {{#fluent.db.is_postgres}}app.databases.use(.postgres(
        hostname: Environment.get("DATABASE_HOST") ?? "localhost",
        port: Environment.get("DATABASE_PORT).flatMap(Int.init(_:)) ?? 5432, // TODO: Add `PostgresConfiguration.ianaPortNumber` like MySQL has
        username: Environment.get("DATABASE_USERNAME") ?? "vapor_username",
        password: Environment.get("DATABASE_PASSWORD") ?? "vapor_password",
        database: Environment.get("DATABASE_NAME") ?? "vapor_database"
    ), as: .psql){{/fluent.db.is_postgres}}{{#fluent.db.is_mysql}}app.databases.use(.mysql(
        hostname: Environment.get("DATABASE_HOST") ?? "localhost",
        port: Environment.get("DATABASE_PORT").flatMap(Int.init(_:)) ?? MySQLConfiguration.ianaPortNumber,
```
Reasons:

- This correctly handles both PostgreSQL and MySQL.
- If the specified port is not a valid integer, this will use the default instead of crashing. This is what happens if any of the other env vars values' are invalid - we don't validate that the host is a valid DNS name or that the database name is a valid Postgres or MySQL identifier, so silently ignoring the port if it's invalid is consistent. It's also much simpler.
- No additional database-specific stanzas are needed.
- With MySQL, the fallback value is not a hardcoded number; the only reason it is for Postgres is I haven't added the exported constant to that driver yet.

Reply:
I think this is a good compromise and cleaner, happy to go with this solution